rllib/agents/tests/test_callbacks.py

```diff
@@ -32,36 +32,41 @@ def tearDownClass(cls):
         ray.shutdown()
 
     def test_on_sub_environment_created(self):
-        config = {
+        base_config = {
             "env": "CartPole-v1",
             # Create 4 sub-environments per remote worker.
             "num_envs_per_worker": 4,
             # Create 2 remote workers.
             "num_workers": 2,
-            "callbacks": OnSubEnvironmentCreatedCallback,
         }
 
-        for _ in framework_iterator(config, frameworks=("tf", "torch")):
-            trainer = dqn.DQNTrainer(config=config)
-            # Fake the counter on the local worker (doesn't have an env) and
-            # set it to -1 so the below `foreach_worker()` won't fail.
-            trainer.workers.local_worker().sum_sub_env_vector_indices = -1
-
-            # Get sub-env vector index sums from the 2 remote workers:
-            sum_sub_env_vector_indices = trainer.workers.foreach_worker(
-                lambda w: w.sum_sub_env_vector_indices
-            )
-            # Local worker has no environments -> Expect the -1 special
-            # value returned by the above lambda.
-            self.assertTrue(sum_sub_env_vector_indices[0] == -1)
-            # Both remote workers (index 1 and 2) have a vector index counter
-            # of 6 (sum of vector indices: 0 + 1 + 2 + 3).
-            self.assertTrue(sum_sub_env_vector_indices[1] == 6)
-            self.assertTrue(sum_sub_env_vector_indices[2] == 6)
-            trainer.stop()
+        for callbacks in (
```

Comment:
adding a test is a great idea!! I have a suggestion though.
we should simply write a separate test, which gets all the "on_xxx" keys from the __dict__ of DefaultCallBacks and MultiCallbacks, and make sure they always match.
that way, we don't have to double check this for every single callback types.
if I read the PR correctly, we are only checking that on_sub_env_creation() callback is defined on both callbacks right now.

Reply:
I add new implementation for that. But the changes do not show up in the GitHub file change page, because this PR was closed, I guess.

Reply:
I opened a new PR #24254.